### PR TITLE
Feat: Bulk mail

### PR DESCRIPTION
**Pull request by 13jksingh**
solves https://github.com/PEC-CSS/acm-website-backend/issues/80

- POST /v1/email/by-role
Sends emails to users based on their roles.

- POST /v1/email/by-user-ids
Sends emails to specific users identified by email IDs.

- POST /v1/email/everyone
Sends emails to all users.

### CODE REVIEW

File: src/main/java/com/pecacm/backend/controllers/EventsController.java
Comment (harasees-singh) [inline src/main/java/com/pecacm/backend/controllers/EventsController.java:1]: please merge main into your branch, this piece of code has been updated

File: src/main/java/com/pecacm/backend/request/EmailRequest.java [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+import lombok.Getter;
+import lombok.Setter;
+
+import java.util.List;
+
+@Getter
+@Setter
+public class EmailRequest {
+    @Nullable
+    private String role;
```
Comment (harasees-singh) [inline src/main/java/com/pecacm/backend/request/EmailRequest.java:13]: can use enum Role

File: src/main/java/com/pecacm/backend/services/EmailService.java [hunk 2]
```diff
@@ -39,4 +49,26 @@ public void sendVerificationEmail(String username) {
...
+            MimeMessageHelper helper;
+            helper = new MimeMessageHelper(message, true);
+            helper.setTo(recipients);
+            helper.setText(htmlBody, true);
+            javaMailSender.send(message);
+        } catch (MessagingException ex) {
+            throw new AcmException("Not able to send mail");
+        }
+    }
+    private String convertMarkdownToHtml(String markdownContent) {
```
Comment (harasees-singh) [inline src/main/java/com/pecacm/backend/services/EmailService.java:68]: please test `convertMarkdownToHtml` thoroughly before merging PR, acm ki izzat ka sawaal hai lol

**Review by harasees-singh — CHANGES_REQUESTED** [review]

**Comment by 13jksingh** [comment]
Post /v1/email/by-user-ids
`{
  "emails": ["harshpreetsinghjohar.btcse20@pec.edu.in","manjotsinghoberoi.btcse20@pec.edu.in"],
  "subject": "Exciting News!",
  "body": "ACM Logo\n\n# Exciting Announcement\n\nDear team,\n\nWe are thrilled to announce a major milestone for our company! With your hard work and dedication, we have achieved great success.\n\nHere's a glimpse of our achievement:\n\nSuccess Graph\n\nThank you for your continued efforts. Let's celebrate this achievement together!\n\nBest regards,\nJaskeerat Singh"
}
`
You must have received the mails for this.
kindly review this @harshjohar @ken1000minus7

**Review by ken1000minus7 — COMMENTED** [review]
Yes received the mail. Btw can't we use a single endpoint for this purpose? If emails list is empty then send email to all and if role is null send email to all roles, if both are empty then send email to everyone 🤔  @harasees-singh @harshjohar

**Review by harasees-singh — APPROVED** [review]

**Review by harshjohar — APPROVED** [review]
good work❤️

**Pull request merged**